### PR TITLE
add allow-privileged flag for FROM DOCKERFILE command

### PR DESCRIPTION
**Pull request by dustyhorizon**
This PR introduces the `--allow-privileged` flag to `FROM DOCKERFILE` command.

Fixes https://github.com/earthly/earthly/issues/3706 although not extensively tested yet but seems to be working thus far.

### CODE REVIEW

**Comment by alexcb** [comment]
Thanks for the PR, I kicked off a test via #3818 (unfortunately the Approve and run github feature doesn't work with our setup).

For this to be merged in, we'll need to create a new feature flag for the `VERSION` keyword -- here's an example of a similar change when the `--network=none` feature was added: https://github.com/earthly/earthly/commit/2721c63ee6d923d6d801606913de47a90117c697#diff-acbd3cf883cc5793dc1ab8d931dd1c5805e6c80785fe4b9979f415087b04e0d9R35

additionally, can we get a test written for this? You could create a new target under https://github.com/earthly/earthly/blob/34e5a37c0f2ed6f7f21ba9374077d43aa630acd7/tests/dockerfile/Earthfile and under the Dockerfile you could run some commands like we do under https://github.com/earthly/earthly/blob/34e5a37c0f2ed6f7f21ba9374077d43aa630acd7/tests/privileged.earth

**Review by alexcb — COMMENTED** [review]
These changes look good, see my previous comment on what's needed to get this PR merged.

**Comment by dustyhorizon** [comment]
@alexcb Thanks for the review and inputs, will work out something!

**Comment by dustyhorizon** [comment]
@alexcb I went ahead and added the feature flag and tests per your comment, do let me know if there are any further issues!

File: tests/Earthfile [hunk 1]
```diff
@@ -513,6 +513,8 @@ dockerfile-test:
         --mount=type=tmpfs,target=/tmp/earthly \
         -- --no-output +test
 
+    DO +RUN_EARTHLY --earthfile=dockerfile/privileged.earth --extra_args="--allow-privileged --no-output" --target=+test-allow-privileged-with-privileged-run-command
```
Comment (alexcb) [inline tests/Earthfile:516]: I think all that's needed is to change the `+test` on line 514 to `+test-all`, then delete this line.

File: earthfile2llb/interpreter.go [hunk 2]
```diff
@@ -806,6 +806,12 @@ func (i *Interpreter) handleFromDockerfile(ctx context.Context, cmd spec.Command
 	if len(args) < 1 {
 		return i.errorf(cmd.SourceLocation, "invalid number of arguments for FROM DOCKERFILE")
 	}
+
+	if !i.converter.ftrs.AllowPrivilegedFromDockerfile && opts.AllowPrivileged {
+		return i.errorf(cmd.SourceLocation, "the FROM DOCKERFILE --allow-privileged flag must be enabled with the VERSION --allow-privileged-from-dockerfile feature flag.")
+	}
+	allowPrivileged := opts.AllowPrivileged && i.allowPrivileged
```
[the remark this answers is not in the document]
Comment (alexcb) [reply earthfile2llb/interpreter.go:813]: sorry, I take this back. this appears to be consistent with the way `FROM --allow-privileged` works.

**Review by alexcb — COMMENTED** [review]
This is looking great.

once the minor comment re tests/Earthfile is addressed I can kick of a test.

File: tests/Earthfile (hunk 1 above)
Comment (alexcb) [reply tests/Earthfile:517]: can this line be deleted?

Comment (dustyhorizon) [reply tests/Earthfile]: Done. Just realized that with the latest fix I don't have to abstract the test anymore :sweat_smile:

**Review by alexcb — APPROVED** [review]

**Comment by alexcb** [comment]
Thanks for the contribution!

**Pull request merged**